common/lua-system.cpp

```diff
@@ -62,10 +69,36 @@ namespace tec {
 		}
 	}
 
+	void LuaSystem::On(std::shared_ptr<ClientConnectedEvent> data) {
+		CallFunction("onClientConnected");
+	}
+	
+	void LuaSystem::On(std::shared_ptr<ClientDisconnectedEvent> data) {
+		CallFunction("onClientDisconnected");
+	}
+
+	void LuaSystem::CallFunction(std::string function_name) {
+		// multiton <eid, LuaScript*>
+		for (auto itr = LuaScriptMap::Begin(); itr != LuaScriptMap::End(); itr++) {
+			if (!itr->second->script_name.empty() && itr->second->environment[function_name].valid()) {
+				itr->second->environment[function_name]();
+			}
+		}
+		// list<scripts>
+		for (auto script = scripts.begin(); script != scripts.end(); script++) {
```

Comment:
So these scripts are sort of the "global" scripts and aren't bound to an entity?

Reply:
yes, at the moment